File: latihan/Latihan 1/Kendaraan.cpp
#include <iostream>
#include <vector>
#include "Kendaraan.hpp"

using namespace std;

/* CTOR */
Kendaraan::Kendaraan(){
    nomor = 0;
    tahunKeluaran = 0;
    merk = "XXX";
    kategori = "mobil";
}

/* CTOR */
Kendaraan::Kendaraan(int nomor, int tahunKeluaran, string merk, string kategori){
    this->nomor = nomor;
    this->tahunKeluaran = tahunKeluaran;
    this->merk = merk;
    this->kategori = kategori;
}

/* CCTOR */
Kendaraan::Kendaraan(Kendaraan &other){
    this->nomor = other.nomor;
    this->tahunKeluaran = other.tahunKeluaran;
    this->merk = other.merk;
    this->kategori = other.kategori;
}

/* Copy Asignment */
Kendaraan Kendaraan::operator=(Kendaraan &other){
    return Kendaraan(other);
}

/* DTOR */
Kendaraan::~Kendaraan(){}

void Kendaraan::printInfo(){
    cout << "Nomor: " << nomor << endl;
    cout << "Merk: " << merk << endl;
    cout << "Kategori: " << kategori << endl;
    cout << "Tahun Keluaran: " << tahunKeluaran << endl;
}

int Kendaraan::biayaSewa(int lamaSewa){
    if(kategori == "bus"){
        return 1000000 * lamaSewa;
    } else if(kategori == "minibus"){
        if(lamaSewa <= 5){
            return 5000000;
        } else{
            return 5000000 + 500000*(lamaSewa-5);
        }
    } else if(kategori == "mobil"){
        return 500000 * lamaSewa;
    } else {
        cout << "Kategori mobil salah!" << endl;
        return 0;
    }
}

void Kendaraan::setData(int nomor, int tahunKeluaran, string kategori, string merk){
    this->nomor = nomor;
    this->tahunKeluaran = tahunKeluaran;
    this->merk = merk;
    this->kategori = kategori;
}

File: latihan/latihan-2/Kendaraan.java
public abstract class Kendaraan {
    protected int nomor;
    protected int tahunKeluaran;
    protected String kategori;
    protected String merk;

    /* CTOR */
    public Kendaraan(){
        nomor = 0;
        tahunKeluaran = 0;
        merk = "XXX";
        kategori = "mobil";
    }

    /* CTOR User Defined */
    public Kendaraan(int _nomor, int _tahunKeluaran, String _merk, String _kategori){
        nomor = _nomor;
        tahunKeluaran = _tahunKeluaran;
        merk = _merk;
        kategori = _kategori;
    }

    /* CCTOR */
    public Kendaraan(Kendaraan other){
        this(other.nomor, other.tahunKeluaran, other.merk, other.kategori);
    }

    public void printInfo(){
        System.out.println("Nomor: " + nomor);
        System.out.println("Merk: " + merk);
        System.out.println("Kategori: " + kategori);
        System.out.println("Tahun Keluaran: " + tahunKeluaran);
    }

    abstract int biayaSewa(int lamaSewa);
}